File: webapi/dal/DataObjects/Game.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using crosscuttingconcerns.Generics;

namespace dal.DataObjects
{
    [Table("Game", Schema = "GameDB")]
    public class Game: IEntity
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        [ForeignKey("GameType")]
        public int GameTypeId { get; set; }
        public GameType GameType { get; set; } = default!;
    }
}

File: webapp/src/app/models/game.ts
import { IGameType, GameType } from './gameType';

export interface IGame {
  id: number;
  name: string;
  description: string;
  gameTypeId: number;
  gameType: IGameType;
}

export class Game implements IGame {
  id!: number;
  name!: string;
  description!: string;
  gameTypeId!: number;
  gameType!: GameType;
}
